zerver/lib/user_topics.py

```diff
@@ -53,20 +54,19 @@ def get_user_topics(
 def get_topic_mutes(
     user_profile: UserProfile, include_deactivated: bool = False
 ) -> List[Tuple[str, str, float]]:
-    query = UserTopic.objects.filter(user_profile=user_profile, visibility_policy=UserTopic.MUTED)
-    # Exclude muted topics that are part of deactivated streams unless
-    # explicitly requested.
-    if not include_deactivated:
-        query = query.filter(stream__deactivated=False)
-    rows = query.values(
-        "stream__name",
-        "topic_name",
-        "last_updated",
+
+    user_topics = get_user_topics(
+        user_profile=user_profile,
+        include_deactivated=include_deactivated,
+        visibility_policy=UserTopic.MUTED,
     )
-    return [
-        (row["stream__name"], row["topic_name"], datetime_to_timestamp(row["last_updated"]))
-        for row in rows
-    ]
+
+    result = []
+    for user_topic in user_topics:
+        stream = access_stream_by_id(user_profile, user_topic["stream_id"])[0]
```

Comment:
Why do we need `access_stream_by_id` here?  It is an expensive query; is there a security reason for it, or are you just trying to get the stream names to preserve the API here?  

If we're going for stream names, the right thing to do is include `stream__name` in what we fetch.

Reply:
Do you mean fetching `stream__name` when `get_user_topics` gets called by `get_muted_topics` or fetching them always inside `get_user_topics` ?
I've updated the PR implementing the former approach. We can do the latter if that offers more utility.